### PR TITLE
Fixed bug with dropdown node when toggling multi selection

### PR DESCRIPTION
**Pull request by fakob**
Fixes this issue - https://trello.com/c/DpnYhJBz

### CODE REVIEW

File: src/TestController.ts [hunk 1]
```diff
@@ -39,14 +37,20 @@ export default class TestController {
...
+    node2Socket: string,
+  ) {
     const n1 = this.getNodeByID(node1ID);
     const n2 = this.getNodeByID(node2ID);
     const originSocket = n1.getOutputSocketByName(node1Socket);
-    PPGraph.currentGraph.connect(
-      originSocket,
-      n2.getSocketForNewConnection(originSocket),
-    );
+    const targetSocket =
```
Comment (magnificus) [inline src/TestController.ts:49]: this is fine but could be made more elegant avoid undefined if we instead make another function that takes 4 inputs, then the function that takes 3 input creates the last argument via n2.getSocketForNewConnection(originSocket), and calls the function with 4 arguments

File: cypress/e2e/pnp/testWidgetNodes.cy.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import { doWithTestController } from './helpers';
+
+describe('testWidgetNodes', () => {
```
Comment (magnificus) [inline cypress/e2e/pnp/testWidgetNodes.cy.ts:3]: very nice with test

**Review by magnificus — APPROVED** [review]

File: src/TestController.ts (hunk 1 above)
Comment (fakob) [reply src/TestController.ts:49]: Sure I can do that, but if I understood correctly, would end up with lot's of duplicated code:
``` javascript
connectNodesByID(node1ID, node2ID, node1Socket, node2Socket) {
    const n1 = this.getNodeByID(node1ID);
    const n2 = this.getNodeByID(node2ID);
    const originSocket = n1.getOutputSocketByName(node1Socket);
    const targetSocket = n2.getInputSocketByName(node2Socket);

    PPGraph.currentGraph.connect(originSocket, targetSocket);
}

connectNodesByIDWithAutoTargetSocket(node1ID, node2ID, node1Socket) {
    const n1 = this.getNodeByID(node1ID);
    const n2 = this.getNodeByID(node2ID);
    const originSocket = n1.getOutputSocketByName(node1Socket);
    const targetSocket = n2.getSocketForNewConnection(originSocket);

    this.connectNodesByID(node1ID, node2ID, node1Socket, targetSocket);
}
```
@TobiasElinder better or did you mean something else?

Comment (magnificus) [reply src/TestController.ts:49]: Ah, yes almost, just the difference is that the one with 3 arguments would call the function with 4 arguments, then you shouldnt have any copy pasted code, so

connectNodesByIDWithAutoTargetSocket
calls
connectNodesByID

by creating the last argument

Comment (fakob) [reply src/TestController.ts:49]: That is what I do above :-)
The problem is that targetSocket for the one with 3 needs to be constructed.

Comment (magnificus) [reply src/TestController.ts:49]: Hmm lets talk about this IRL today

Comment (magnificus) [reply src/TestController.ts:49]: aha I understand what you are saying now, right yes fair enough lets do it as you did first.

**Pull request merged**